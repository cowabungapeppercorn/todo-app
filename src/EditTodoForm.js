import React from 'react';

class EditTodoForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = { 
      text: this.props.text
      // editMode: false
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(evt) {
    evt.preventDefault();
    this.props.editTodo(this.props.id, this.state.text);
  }

  handleChange(evt) {
    this.setState({ [evt.target.name]: evt.target.value });
  }

  render() {
    return (
      <div>
        <form onSubmit={this.handleSubmit}>

          <label htmlFor="text">Todo:</label>
          <input id="text" name="text" type="text"
            value={this.state.text}
            onChange={this.handleChange} />

          <button type="submit">Edit</button>

        </form>
      </div>
    );
  }
}

export default EditTodoForm; 