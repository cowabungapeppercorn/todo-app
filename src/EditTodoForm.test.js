import React from 'react';
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json';
import EditTodoForm from './EditTodoForm';

it('renders without crashing', () => {
  shallow(<EditTodoForm />);
});

it('renders as expected', () => {
  let wrapper = mount(<EditTodoForm text='taco'/>);
  wrapper.instance().handleChange({
    target: {
      name: "text",
      value: "fix this bug"
    }
  });
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});