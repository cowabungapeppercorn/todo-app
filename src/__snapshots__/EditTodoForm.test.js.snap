// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<EditTodoForm
  text="taco"
>
  <div>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="text"
      >
        Todo:
      </label>
      <input
        id="text"
        name="text"
        onChange={[Function]}
        type="text"
        value="taco"
      />
      <button
        type="submit"
      >
        Edit
      </button>
    </form>
  </div>
</EditTodoForm>
`;
